### ISSUE TITLE
`swaymsg exec` breaks bzip2 pipe on foot and st.

### ISSUE THREAD

**Issue by crocket** [swaywm/sway#6544]
- **Sway Version:**
sway version 1.6.1

- **Debug Log:**
[sway.log](https://github.com/swaywm/sway/files/7218461/sway.log)

- **Configuration File:**

```
set $mod Mod4

bindsym $mod+Return exec st
```

- **Description:**

If I execute `man man` on `st` executed by `$mod+Return` or `swaymsg exec st`, I get

```
bzip2: I/O or other error, bailing out.  Possible reason follows.
bzip2: Broken pipe
	Input file = (stdin), output file = (stdout)

bzip2: I/O or other error, bailing out.  Possible reason follows.
bzip2: Broken pipe
	Input file = (stdin), output file = (stdout)
```

I don't see this error if I execute `st` without using sway's exec command.

This issue is reproducible on `foot` terminal emulator.

Alacritty emulator doesn't have this issue.

**Comment by RPigott** [comment]
Sway just execs what you tell it to and records the pid. It doesn't mess with your environment in a way that could affect this. This is not a sway issue.

**Comment by crocket** [comment]
I also read it before submitting this issue. There needs to be a fix. Either, someone creates a bzip2 fork, or someone finds a workaround. The root problem may not necessarily be bzip2's fault.

**Comment by eonpatapon** [comment]
Same issue here,

When foot is exec by sway I get a lot of Broken pipe errors on the command line.

For example:

``
cat /dev/urandom | head -c2
at: write error: Broken pipe
```

This doesn't happen when foot is run from wofi.

It doesn't happen at all with terminals like urxvt or kitty

Maybe relevant: https://unix.stackexchange.com/questions/588123/broken-pipe-works-differently-in-different-terminal-emulators

**Comment by dnkl** [comment]
> Maybe relevant: https://unix.stackexchange.com/questions/588123/broken-pipe-works-differently-in-different-terminal-emulators

One comment says this:

> So it's your window manager (or whatever app/menu/file manager you're starting st from) which is ignoring the PIPE signal (and st and the shell are inheriting the disposition). lilyterm is probably resetting it to default. That's a bug in the window manager -- it shouldn't be doing that.

That _would_ explain what we're seeing here. Foot currently does **not** reset signal disposition (though it's being discussed to change this), but I believe Kitty does, and wouldn't be surprised if urxvt does it too.

**Comment by dnkl** [comment]
(and to make it clear - if this is indeed what's happening, then it _is_ a Sway bug)

**Comment by eonpatapon** [comment]
@dnkl looks like you already fixed this https://github.com/swaywm/sway/commit/1dd6df6a5df3924d42a25bd17dab0ef8beecad61 :)
Unfortunately I'm still on 1.6.

**Comment by dnkl** [comment]
> @dnkl looks like you already fixed this in 1dd6df6 :)

So I did... Had completely forgotten 🤦
